### DISCUSSION TITLE
Questions about mapping reconstructed volume to the reference frame

### DISCUSSION THREAD

**Discussion by aliciachenw** [PlusToolkit/PlusLib#977] [Q&A]
Hi! Many thanks for this wonderful project. I have a question about the coordinate system of the reconstructed US volume: after I finish the tracked freehand US scanning and the volume reconstruction using VolumeReconstructor, how can I map the US volume back to the reference coordinate frame? 
I noticed that in the reconstructed volume header, there are two attributes "Offset" and "TransformMatrix", and the user manual said "the volume will be reconstructed in reference coordinate system". So I am wondering if it means that I can simply add the offset to transform the volume to the reference coordinate system.
Thanks very much!

**Comment by ungi** [comment]
Hi, although it works, I think it's not recommended to use the offset and transform matrix in the volume node. It's better to create a separate transform node and set that as the transform of the reconstructed volume. If the input images were already on a transform node e.g. called ImageToReference, then your volume will automatically be in the Reference coordinate system. If you have something like ReferenceToPatient, then you can apply that on the output volume so it will be transformed to the Patient coordinate system.

**Reply by aliciachenw** [reply]
Hi Tamas, thank you! Yes, I am thinking of adding another node to do the transformation, just want to confirm if this is the transformation that I am looking for. Many thanks for answering!
